### PR TITLE
[MM-14081] Remove checkMFA invocations and replace with error-based MFA handling

### PR DESCRIPTION
**Pull request by DSchalla**
#### Summary
This PR changes the MFA authentication flow to check whether MFA is required using an error-based approach where first an login is attempted without no MFA token, and if a specific error is returned the MFA form is shown. The overall goal is to avoid leaking the MFA state of a user without having any sort of credentials. Please see the server PR for my details.

#### Ticket Link
https://mattermost.atlassian.net/projects/MM/issues/MM-14081

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Needs to be implemented in mobile
- [x] Has server changes: https://github.com/mattermost/mattermost-server/pull/10356
- [x] Touches critical sections of the codebase (auth, posting, etc.)

### CODE REVIEW

File: actions/views/mfa.js [hunk 1]
```diff
@@ -29,6 +29,9 @@ export function generateMfaSecret() {
     };
 }
 
+/**
+ * @deprecated Clients should attempt to login without MFA and check if MFA token error is returned to show MFA prompt
```
Comment (hmhealey) [inline actions/views/mfa.js]: Do we use `@deprecated` elsewhere? I'm not sure if we have a consistent way of marking deprecated methods in comments.

Alternatively, this is just part of the web app, so we should be able to completely remove this function if we're not using it any more

File: components/claim/components/email_to_oauth.jsx [hunk 2]
```diff
@@ -74,15 +59,20 @@ export default class EmailToOAuth extends React.PureComponent {
...
+                }
             }
         );
     }
 
     render() {
         var error = null;
         if (this.state.error) {
-            error = <div className='form-group has-error'><label className='control-label'>{this.state.error}</label></div>;
+            error =
```
Comment (hmhealey) [inline components/claim/components/email_to_oauth.jsx]: If we're breaking this into two lines, we'd generally surround the JSX with brackets. I think that might be enforced by the style checker.

File: actions/views/mfa.js (hunk 1 above)
Comment (DSchalla) [reply actions/views/mfa.js]: Removing it completely sounds good to me.

File: components/claim/components/email_to_oauth.jsx (hunk 2 above)
Comment (DSchalla) [reply components/claim/components/email_to_oauth.jsx]: Yup, that was an auto-format mistake from the IDE. I'll change that back.

**Pull request merged**